spec/alephant/logger/json_spec.rb

```diff
@@ -1,137 +1,123 @@
+            let(:write_level) { logging_levels.index(level) }
 
-    it 'provides a static method for checking if a session has been set' do
-      described_class.session = fn
-      expect(described_class.session?).to eq 'instance-variable'
+            it_behaves_like 'a JSON log writer'
 
-      described_class.send(:remove_instance_variable, :@session)
-      expect(described_class.session?).to eq nil
-    end
-  end
+            it_behaves_like 'nested JSON message flattened to strings'
 
-  shared_context 'nested log hash' do
-    let(:log_hash) do
-      { 'nest' => nest }
-    end
+            it_behaves_like 'gracefully fails with string message'
+          end
+        end
 
-    let(:nest) { { 'bird' => 'eggs' } }
-  end
+        context 'when message level is lower than write level' do
+          context 'Integer' do
+            let(:write_level) { logging_levels.index(level) + 1 }
 
-  shared_examples 'nests flattened to strings' do
-    include_context 'nested log hash'
+            it_behaves_like 'a JSON log non writer'
 
-    specify do
-      expect(log_file).to receive(:write) do |json_dump|
-        expect(JSON.parse(json_dump)['nest']).to eq nest.to_s
-      end
+            it_behaves_like 'gracefully fails with string message'
+          end
+        end
 
-      subject.send(level, log_hash)
-    end
-  end
+        context 'when message level is higher than write level' do
+          let(:write_level_index) do
+            i > 0 ? logging_levels.index(level) - 1 : 0
+          end
 
-  shared_examples 'nesting allowed' do
-    include_context 'nested log hash'
+          context 'Symbol' do
+            let(:write_level) { logging_levels[write_level_index] }
 
-    specify do
-      expect(log_file).to receive(:write) do |json_dump|
-        expect(JSON.parse(json_dump)).to eq log_hash
-      end
+            it_behaves_like 'a JSON log writer'
 
-      subject.send(level, log_hash)
-    end
-  end
+            it_behaves_like 'nested JSON message flattened to strings'
 
-  shared_examples 'gracefully fail with string arg' do
-    let(:log_message) { 'Unable to connect to server' }
+            it_behaves_like 'gracefully fails with string message'
+          end
 
-    specify { expect(log_file).not_to receive(:write) }
-    specify do
-      expect { subject.debug log_message }.not_to raise_error
-    end
-  end
+          context 'String' do
+            let(:write_level) { logging_levels[write_level_index].to_s }
 
-  %w(debug info warn error).each do |level|
-    describe "##{level}" do
-      let(:level) { level }
+            it_behaves_like 'a JSON log writer'
 
-      it_behaves_like 'JSON logging'
+            it_behaves_like 'nested JSON message flattened to strings'
 
-      it_behaves_like 'nests flattened to strings'
+            it_behaves_like 'gracefully fails with string message'
+          end
 
-      it_behaves_like 'gracefully fail with string arg'
+          context 'Symbol' do
```

Comment:
Seem to have symbol in twice.

Reply:
oh, good call. Meant to be an integer. 👍